app/src/main/java/org/dhis2/usescases/main/MainNavigator.kt

```diff
@@ -133,38 +137,42 @@ class MainNavigator(
             onTransitionStart()
             currentScreen.value = screen
             currentFragment = fragment
-            val transaction: FragmentTransaction = fragmentManager.beginTransaction()
-            transaction.apply {
-                if (sharedView == null) {
-                    val (enterAnimation, exitAnimation) = if (useFadeInTransition) {
-                        Pair(android.R.anim.fade_in, android.R.anim.fade_out)
-                    } else {
-                        Pair(R.anim.fragment_enter_right, R.anim.fragment_exit_left)
-                    }
-                    val (enterPopAnimation, exitPopAnimation) = if (useFadeInTransition) {
-                        Pair(android.R.anim.fade_in, android.R.anim.fade_out)
-                    } else {
-                        Pair(R.anim.fragment_enter_left, R.anim.fragment_exit_right)
+
+            CoroutineScope(Dispatchers.Main).launch {
```

Comment:
Why using the dispatchers like this? Shouldn't they be parameters in the constructor?

Reply:
Yes it was flagged by Sonar, however I did not want to include many changes this close to the release. Anyway I have introduced a flaky test with this so I will have to review this anyway

Reply:
I have updated the PR